contracts/PodsUpkeep.sol

```diff
@@ -54,66 +48,47 @@ contract PodsUpkeep is KeeperCompatibleInterface, Ownable {
         
         address[] memory pods = podsRegistry.getAddresses();
         for(uint256 i = 0; i < pods.length; i++){ // can get out of gas here -- do we want to implement a batchLimit ?
-            if(checkUpkeepRequired(IPod(pods[i]))){
+            if(checkUpkeepRequired(i)){
                 return (true, "");
             }
         }
         return (false, "");    
     }
 
-    /// @notice Performs upkeep on the pods contract
+    /// @notice Performs upkeep on the pods contract and updates lastUpkeepBlockNumbers
     /// @param performData Not used in this implementation.
     function performUpkeep(bytes calldata performData) override external {
     
         address[] memory pods = podsRegistry.getAddresses();
+        
         for(uint256 i = 0; i < pods.length; i++){
-            (bool required, uint256 batchAmount) = amountAboveFloat(IPod(pods[i])); 
+            
+            bool required = checkUpkeepRequired(i);
+            
             if(required) {
-                require(IPod(pods[i]).batch(batchAmount), "PodsUpkeep: batch() failed");
+                require(IPod(pods[i]).batch(), "PodsUpkeep: batch() failed");
+                lastUpkeepBlockNumbers.lastPodUpkeep[i] = uint32(block.number);
```

Comment:
This isn't going to save gas, unfortunately.

The entire array has to be written *all at once*.   This is why it's better as a single uint256